packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -554,11 +556,12 @@ class IOSSimulator extends Device {
   String get sdkNameAndVersion => category;
 
   @override
-  DeviceLogReader get logReader {
-    if (_logReader == null)
-      _logReader = new _IOSSimulatorLogReader(this);
+  DeviceLogReader get logReader => logReaderForApp(null);
```

Comment:
Here and elsewhere: bit unexpected that logReaderForApp supports `null`. Maybe just return it directly? Not the end of the world either way though.